build/common.sh

```diff
@@ -582,462 +567,154 @@ function kube::build::run_build_command() {
-      if [[ "${platform%/*}" == "windows" ]]; then
-        client_bins=("${KUBE_CLIENT_BINARIES_WIN[@]}")
-      fi
-
-      # This fancy expression will expand to prepend a path
-      # (${LOCAL_OUTPUT_BINPATH}/${platform}/) to every item in the
-      # KUBE_CLIENT_BINARIES array.
-      cp "${client_bins[@]/#/${LOCAL_OUTPUT_BINPATH}/${platform}/}" \
-        "${release_stage}/client/bin/"
-
-      kube::release::clean_cruft
-
-      local package_name="${RELEASE_DIR}/kubernetes-client-${platform_tag}.tar.gz"
-      kube::release::create_tarball "${package_name}" "${release_stage}/.."
-    ) &
+  local tries=10
+  while (( ${tries} > 0 )) ; do
+    if ${netcat} -z "$1" "$2" 2> /dev/null ; then
+      return 0
+    fi
+    tries=$(( ${tries} - 1))
+    sleep 0.1
   done
 
-  kube::log::status "Waiting on tarballs"
-  kube::util::wait-for-jobs || { kube::log::error "client tarball creation failed"; exit 1; }
+  return 1
 }
 
-# Package up all of the server binaries
-function kube::release::package_server_tarballs() {
-  local platform
-  for platform in "${KUBE_SERVER_PLATFORMS[@]}"; do
-    local platform_tag=${platform/\//-} # Replace a "/" for a "-"
-    local arch=$(basename ${platform})
-    kube::log::status "Building tarball: server $platform_tag"
-
-    local release_stage="${RELEASE_STAGE}/server/${platform_tag}/kubernetes"
-    rm -rf "${release_stage}"
-    mkdir -p "${release_stage}/server/bin"
-    mkdir -p "${release_stage}/addons"
-
-    # This fancy expression will expand to prepend a path
-    # (${LOCAL_OUTPUT_BINPATH}/${platform}/) to every item in the
-    # KUBE_SERVER_BINARIES array.
-    cp "${KUBE_SERVER_BINARIES[@]/#/${LOCAL_OUTPUT_BINPATH}/${platform}/}" \
-      "${release_stage}/server/bin/"
-
-    kube::release::create_docker_images_for_server "${release_stage}/server/bin" "${arch}"
-
-    # Include the client binaries here too as they are useful debugging tools.
-    local client_bins=("${KUBE_CLIENT_BINARIES[@]}")
-    if [[ "${platform%/*}" == "windows" ]]; then
-      client_bins=("${KUBE_CLIENT_BINARIES_WIN[@]}")
-    fi
-    cp "${client_bins[@]/#/${LOCAL_OUTPUT_BINPATH}/${platform}/}" \
-      "${release_stage}/server/bin/"
-
-    cp "${KUBE_ROOT}/Godeps/LICENSES" "${release_stage}/"
-
-    cp "${RELEASE_DIR}/kubernetes-src.tar.gz" "${release_stage}/"
+# This is the port that rsync is running on *inside* the container. This may be
+# mapped to KUBE_RSYNC_PORT via docker networking.
+readonly KUBE_CONTAINER_RSYNC_PORT=8730
 
-    kube::release::clean_cruft
-
-    local package_name="${RELEASE_DIR}/kubernetes-server-${platform_tag}.tar.gz"
-    kube::release::create_tarball "${package_name}" "${release_stage}/.."
-  done
-}
+function kube::build::start_rsyncd_container() {
+  kube::build::stop_rsyncd_container
+  V=6 kube::log::status "Starting rsyncd container"
+  kube::build::run_build_command_ex \
+    "${KUBE_RSYNC_CONTAINER_NAME}" -p 127.0.0.1:${KUBE_RSYNC_PORT}:${KUBE_CONTAINER_RSYNC_PORT} -d \
```

Comment:
Yeah -- that causes docker to pick an ephemeral port.  We query that with `docker port` below.